rust/kernel/sync/semaphore.rs

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! semaphore
+//!
+//! C header: [`include/linux/semaphore.h`](../../../../include/linux/semaphore.h)
+
+// To be removed once updated to the latest version of the Linux.
```

Comment:
We won't remove it. Add a TODO to indicate we need to refactor this to rust-style.